CIPs/CIP-82/CIP-82.md

```diff
@@ -70,25 +67,23 @@ This way, by loading the `Notes` schema, it is possible by a tool/library to dis
 
 ## Specification
 
-References to Ceramic schema should use a `string` with the `$ceramic` field using the `tile` type, and optionally with a `schema` property containing either a string or array of strings containing the DocID (implicit reference to latest version) or CommitID (specific version) of the supported schema(s):
+References to Ceramic schema should use a `string` with the `$comment` field using the `ceramic:tile` type, and optionally with a schema string of the StreamID (implicit reference to latest version) or CommitID (specific version) of the supported schema(s).
+Multiple schemas can be provided, using the `|` character as separator.
 
 ```js
 {
   type: 'string',
   maxLength: 150,
-  $ceramic: {
-    type: 'tile',
-    schema: '<Note schema docID>',
-  },
+  $comment: 'ceramic:tile:<Note schema streamID>',
```

Comment:
A part of me wonders if we'd be better off using the streamtype code rather than name (so 0 for tile, 1 for caip10link, etc). I'm worried that once we have a large number of stream types that all represent json documents just with different log structures, coming up with meaningful human readable names for them all is going to get tricky...

Reply:
Yes I agree there might be a better namespace than `tile` but I'm not sure stream types are so relevant here, because the purpose is to interact with other JSON documents anyways?
At this level I don't think it matters much if the JSON doc is backed by a tile, CRDTs or any other stream type, but rather than it can be resolved by the Ceramic client, no?

Reply:
ah right, good point.  In that case, is there any reason to include the streamtype name at all?  Could it just be `ceramic:<streamID>`?  The StreamID already has the streamtype ID number embedded within it.

Reply:
`tile` here was meant as a namespace, as we have `ceramic:appendCollection:<slice schema ID>` and `ceramic:collectionSlice` that have different meaning.
Basically `ceramic:tile` should just mean "this string points to a Ceramic JSON document" and `ceramic:tile:<schema ID>` means a reference to a JSON doc having a specific schema.
Maybe we can just use `ceramic:doc` or `ceramic:ref` instead if it is more explicit?

Reply:
What's gained by this namespacing?  Is it just for human readability? Or does the user or code many any decisions based on the namespace?

Reply:
Purely tooling at the moment for docset and GraphQL support purposes. I don't think human readability matters too much, at least it's not meant for end-users, but might be simpler for debugging.

Reply:
okay, so there *is* code that is going to make decisions based on this namespace?  Can you clarify where/how?

Reply:
Sure, you can see the tile reference being used with docsets to extract schema references here: https://github.com/ceramicstudio/js-idx/blob/feat/append-collection/packages/tools/src/docset.ts#L46-L56
Similar here for an append collection to extract the slice schema: https://github.com/ceramicstudio/js-idx/blob/feat/append-collection/packages/append-collection/src/append-collection.ts#L39-L47

Reply:
Okay, so it seems like it's mostly for sanity check input validation.  Nothing intrinsically that couldn't be done just with `ceramic:<streamid>`, but just adds some extra validation that the right types of streams are being used in the right places.

Something like `ceramic:doc` seems like it would make sense to me, but it kind of depends on some decisions we haven't really made yet on what the relationships will be between different streamtypes that implement the same user-facing APIs (e.g. different stream types that all represent a json document, but with different log structures, conflict resolution mechanisms, history retention, etc).

Reply:
My current vote is for `ceramic:doc:<streamID>`, with the understanding that `ceramic:doc` refers to a streamtype that is one of the streamtypes within the group of streams that expose a JSON document API.

How client/application code will know what is the set of streamtypes that map to that type of API is still an open question however

Reply:
👍 I replaced `tile` by `doc`

Reply:
I don't really like that we invent a new format of encoding streamids here.

I propose that we use the standard url for streamids with an additional query param:
- `ceramic://<streamid>?json-schema=doc`
- `ceramic://<streamid>?json-schema=appendCollection`

Reply:
These specs are about metadata in the schemas themselves, not about the contents.
In `ceramic:doc:<id>` the `id` is the stream ID of the schema the referenced doc should use, not the contents doc.

These strings don't even necessarily represent stream IDs, for example `ceramic:appendCollection` is just a way for the tools to detect a schema represents an AppendCollection.

Reply:
I get that, but I don't see why we need to introduce another representation. As far as I can tell we always include the StreamID. Any problem with using the formatting I suggested above so that we don't introduce a new representation?